### DISCUSSION TITLE
Decide which functions to move from experimental to proposed

### DISCUSSION THREAD

**Discussion by krashish8** [pgRouting/pgrouting#2195] [General]
## Experimental Functions

Decide which functions shall be moved to proposed in v3.3.0
(Comment on this discussion and/or edit this)

YES/NO | Family | Function | Doc | No crash test | Types check | Inner query test | Unit tests | Equivalence test | Comments |
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
  | Flow | pgr_maxFlowMinCost | Yes | Yes | Yes | Yes | - | - | - |
  | Flow | pgr_maxFlowMinCost_Cost | Yes | Yes | Yes | Yes | - | - | -
NO  | Chinese Postman Problem | pgr_chinesePostman | Yes | - | Yes | Yes | - | - | should work for directed & undirected graphs currently only works for directed. More tests are needed
NO  | Chinese Postman Problem | pgr_chinesePostmanCost | Yes | - | Yes | Yes | - | - | same as pgr_chinesePostman
YES  | Coloring | pgr_sequentialVertexColoring | Yes | Yes | Yes | Yes | 18 | - | -
  | Coloring | pgr_bipartite | Yes | Yes | Yes | Yes | 16 | - | -
NO | Coloring | pgr_edgeColoring | Yes | Yes | Yes | Yes | 27 | - | Issue about number of colors need to be tested with boost
  | Transformation | pgr_lineGraph | Yes | Yes | Yes | Yes | - | - | -
  | Transformation | pgr_lineGraphFull | Yes | Yes | Yes | Yes | Yes | Yes | -
YES  | Traversal | pgr_depthFirstSearch | Yes | Yes | Yes | Yes | 36 | Yes | -
  | Components | pgr_makeConnected | Yes | Yes | Yes | Yes | 20 | - | -
YES  | Dijkstra | pgr_dijkstraNear | Yes | Yes | Yes | Yes | Yes | Yes | Has additional tests
YES  | Dijkstra | pgr_dijkstraNearCost | Yes | Yes | Yes | Yes | Yes | Yes | Has additional tests
NO  | VRP | pgr_pickDeliver | Yes | - | Yes | Yes | - | - | Further development will happen on vrpRouting
NO  | VRP | pgr_pickDeliverEuclidean | Yes | - | Yes | Yes | - | - | Further development will happen on vrpRouting
NO  | VRP | pgr_vrpOneDepot | Yes | Yes | - | - | - | - | Further development will happen on vrpRouting
  | - | pgr_bellmanFord | Yes | Yes | Yes | Yes | - | Yes | -
No | - | pgr_binaryBreadthFirstSearch | Yes | Yes | Yes | Yes | 18 | Yes | Needs additional roadworks table
  | - | pgr_breadthFirstSearch | Yes | Yes | Yes | Yes | 20 | - | -
  | - | pgr_dagShortestPath | Yes | Yes | Yes | Yes | - | - | -
  | - | pgr_edwardMoore | Yes | Yes | Yes | Yes | - | Yes | -
  | - | pgr_isPlanar | Yes | Yes | Yes | Yes | 16 | - | -
  | - | pgr_stoerWagner | Yes | - | Yes | Yes | - | Yes | -
  | - | pgr_topologicalSort | Yes | Yes | - | Yes | - | - | -
  | - | pgr_transitiveClosure | Yes | Yes | - | Yes | - | - | -
No | - | pgr_turnRestrictedPath | No | - | - | - | - | - | -
  | - | pgr_lengauerTarjanDominatorTree | Yes | Yes | Yes | Yes | 11 | - | -

**Additional Comments**:

- Line Graph: Files are not organized/named properly - no_crash_test of both Line Graph and Line Graph Full are present in the LineGraph directory.
- Shall we ignore the VRP functions, because of the vrpRouting repository?
- All the Combinations SQL functions are also Experimental. Some of them can be moved to proposed. Refer: pgRouting/pgrouting#1648

cc: @cvvergara

**Comment by cvvergara** [comment]
Functions without family/category lets keep experimental

**Comment by cvvergara** [comment]
Move the all combination functions to proposed on 3.3

**Reply by cvvergara** [reply]
Except the ones that belong to groups that are experimental.

**Reply by cvvergara** [reply]
By inspecting the release notes, there is nothing to do, combination functions are well marked
.
